src/commonMain/kotlin/com/charleskorn/kaml/YamlInput.kt

```diff
@@ -62,6 +62,13 @@ public sealed class YamlInput(
             is YamlList -> when (descriptor.kind) {
                 is StructureKind.LIST -> YamlListInput(node, yaml, context, configuration)
                 is SerialKind.CONTEXTUAL -> createContextual(node, yaml, context, configuration, descriptor)
+                is PolymorphicKind -> {
+                    if (descriptor.isContentBasedPolymorphic) {
+                        createContextual(node, yaml, context, configuration, descriptor)
+                    } else {
+                        throw MissingTypeTagException(node.path)
+                    }
+                }
```

Comment:
Why is this change required?

Reply:
To be able to decode a `YamlList` still as `YamlNode`. The `YamlNodeSerializer.descriptor` is polymorphic (because we don't know the type of it). So a list needs to be readable as polymorphic.